packages/discussions/src/utils/channels/can-post-to-channel.ts

```diff
@@ -44,11 +52,148 @@ export function canPostToChannel(
   });
 }
 
+function isAuthorizedToPostByChannelAcl(
+  user: IDiscussionsUser,
+  channelAcl: IChannelAclPermission[]
+): boolean {
+  const permissions = mapPermissionsByCategory(channelAcl);
```

Comment:
I'm kind of thinking we should maybe make this a class and pass the ACL to constructor.  Then have instance method called `validate` or something so we could do:

```js
class ChannelPermission {
  constructor(channelAcl) {
    this.permissions = mapPermissionsByCategory(channelAcl);
  }

  canPost( user) {
    // do stuff
  }

  canModifyPost (user) {
    // do stuff
  }
}

// Usage

const channelPermission = new ChannelPermission(channelAcl)

channelPermission.canPost(user)
```

Reply:
Good idea. Then we can reuse some helper functions